### PR TITLE
fix: repair linotp migrationscript

### LINKED ISSUE #430: tools/edumfa-migrate-linotp.py only working with SQLAlchemy < 2
### Top-level intent

Migration from LinOTP to eduMFA

### Steps to reproduce

1. run tools/edumfa-migrate-linotp.py script
2. get error:
`
Traceback (most recent call last):
  File "/opt/edumfa/tools/./edumfa-migrate-linotp.py", line 586, in <module>
    main()
  File "/opt/edumfa/tools/./edumfa-migrate-linotp.py", line 576, in main
    migrate(config_obj)
  File "/opt/edumfa/tools/./edumfa-migrate-linotp.py", line 351, in migrate
    s = select([realm_table])
...
sqlalchemy.exc.ArgumentError: Column expression, FROM clause, or other columns clause element expected, got [Table('
`

### Expected outcome

Migrate Tokens from LinOTP database to eduMFA database.

### Actual outcome

Error message.

### Configuration

* **eduMFA version**: 2.3.0
* **Installation method**: PIP
* **Python version**: 3.9.2
* **Operating system**: Debian
* **Webserver**: Apache
* **Token database**:  PostgreSQL

### Log file

External tool. No logfile. 

### Workaround

Install second virtual env with eduMFA 2.0.3. I think, this is the last version with SQLAlchemy < 2. Seem like there are some API changes in SQLAlchemy that break the script.

### PR DESCRIPTION
**Pull request by cyber-simon**
Trying to reflect changes, that are needed to get script working again for SQLAlchemy 2.

Mainly a few syntax changes, and behavour change with autocommit. Added support for Token without encryption parameters.

Closes #430

### CODE REVIEW

File: tools/edumfa-migrate-linotp.py [hunk 1]
```diff
@@ -332,7 +333,8 @@ def insert_chunks(conn, table, values, chunk_size=100000, record_name="records")
             print('Insert {} {} to {} ...'.format(record_name, chunk,
                                                   min(chunk + chunk_size, values_length) - 1))
             try:
-                conn.execute(table.insert(), values[chunk:chunk+chunk_size])
+                sess.execute(table.insert(), values[chunk:chunk+chunk_size])
```
Comment (cyber-simon) [inline tools/edumfa-migrate-linotp.py:336]: execute command needs to be on Session now, if not, nothing will be commited.

File: tools/edumfa-migrate-linotp.py [hunk 2]
```diff
@@ -412,7 +414,7 @@ def insert_chunks(conn, table, values, chunk_size=100000, record_name="records")
 
             key_enc = r['LinOtpKeyEnc']
             key_iv = r['LinOtpKeyIV']
-            if config_obj.REENCRYPT:
+            if config_obj.REENCRYPT and key_enc != "" and key_iv != "":
```
Comment (cyber-simon) [inline tools/edumfa-migrate-linotp.py:417]: If REENCRYPT is turned on, the script will fail on token, that have no encryption. Like "yubico". They just have an id which can not be reencrypted.

**Comment by fritterhoff** [comment]
Hi!

thanks for the submission and sorry for the delay.

To be honest we did not check this script since all team members were using PrivacyIdea.

File: tools/edumfa-migrate-linotp.py (hunk 2 above)
Comment (fritterhoff) [reply tools/edumfa-migrate-linotp.py:417]: Looks fine for me

File: tools/edumfa-migrate-linotp.py (hunk 1 above)
Comment (fritterhoff) [reply tools/edumfa-migrate-linotp.py:336]: Prop a change introduced by SQLAlchemy.

**Review by fritterhoff — APPROVED** [review]

**Pull request merged**